### PR TITLE
W-13806048: Ignore Expect: 100-Continue header in server response and log warning

### PR DESCRIPTION
**Pull request by mbuchwald**

### CODE REVIEW

File: modules/http/src/main/java/org/glassfish/grizzly/http/HttpCodecFilter.java [hunk 1]
```diff
@@ -1047,7 +1047,11 @@ private static void finalizeKnownHeaderNames(final HttpHeader httpHeader,
         } else if (size == Header.Expect.getLowerCaseBytes().length) {
             if (ByteChunk.equalsIgnoreCaseLowerCase(input, start, end,
                     Header.Expect.getLowerCaseBytes())) {
-                ((HttpRequestPacket) httpHeader).requiresAcknowledgement(true);
+                if (httpHeader.isRequest()) {
+                    ((HttpRequestPacket) httpHeader).requiresAcknowledgement(true);
+                } else {
+                    LOGGER.warning("asdf");
```
Comment (eze210) [inline modules/http/src/main/java/org/glassfish/grizzly/http/HttpCodecFilter.java]: Print some descriptive message here

**Review by eze210 — CHANGES_REQUESTED** [review]
How complicated is to add a test case at this level?

**Comment by mbuchwald** [comment]
The problem with the test on grizzly side is that implies mocking too many things, in a scenario of a requester that is recieving a header that behaves a ser request header instead of server, so trying to avoid some (other) class cast exceptions make that everything is mocked checking how to avoid those cases as white box test. I think it's way better to have the http service test

**Review by eze210 — APPROVED** [review]
For future reference: https://github.com/mulesoft/mule-http-service/pull/542

**Pull request merged**